src/lmflow/pipeline/finetuner.py

```diff
@@ -311,8 +311,7 @@ def __init__(self, n_layers, interval_steps, model):
                         self.layers_attribute = 'model.transformer.h'  # General access path
                     self.total_layers = len(eval('self.' + self.layers_attribute))  # Dynamically execute to get the number of layers
 
-                    # Freeze all layers upon initialization
-                    self.freeze_all_layers()
+                    self.switch_active_layers()
                     self.active_layers_indices = []
```

Comment:
I think these lines should be swapped, since `active_layers_indices` is modified by `switch_active_layers`